src/Receipts/ReceiptResponse.php

```diff
@@ -98,26 +118,59 @@ public function getIsRetryable(): ?bool
     }
 
     /**
-     * @return string
+     * @return string|null
      */
     public function getLatestReceipt(): ?string
     {
         return $this->latestReceipt;
     }
 
     /**
-     * @return array|ReceiptInfo[]
+     * @return array|LatestReceiptInfo[]
      */
-    public function getLatestReceiptInfo(): array
+    public function getLatestReceiptInfo(): ?array
     {
+        if (is_null($this->latestReceiptInfo)) {
+            return null;
+        }
+
+        if ($this->parsedLatestReceiptInfo) {
+            return $this->latestReceiptInfo;
+        }
+
+        $data = [];
+
+        foreach ($this->latestReceiptInfo as $receiptInfo) {
+            $data[] = LatestReceiptInfo::fromArray($receiptInfo);
+        }
+
+        $this->latestReceiptInfo = $data;
+        $this->parsedLatestReceiptInfo = true;
+
         return $this->latestReceiptInfo;
     }
 
     /**
-     * @return array|PendingRenewal[]
+     * @return array|PendingRenewal[]|null
      */
-    public function getPendingRenewalInfo(): array
+    public function getPendingRenewalInfo(): ?array
```

Comment:
also here seems to be a type mismatch with docblock

Reply:
Explain more please

Reply:
i don't know what i have seen here yesterday 🤦‍♂️ all good